### ISSUE TITLE
change how checkpoint works

### ISSUE THREAD

**Issue by NickleDave** [NickleDave/vak#159]
- [ ] remove "save_only_single_checkpoint_file" option
  + most people won't want to save many checkpoints throughout training
- [ ] instead, save two types of checkpoints:
  1. the standard checkpoint that gets saved intermittently throughout training as a backup, and that gets saved at the end of training to capture the state of the model after the specified number of epochs
  2. a `max-val-acc` checkpoint that only gets saved when validation accuracy improves

The reason for saving two (and only two) checkpoints is that this allows us to just automatically save the checkpoint at the end of training, after iterating through `n` epochs, *without* worrying about whether we are saving over the checkpoint that had the maximum validation accuracy.

Other options would be to modify the logic in the checkpoint loop, either to cleverly check whether we have been saving a checkpoint based on validation accuracy, *or* only save at the end of training if we have *not* been saving a checkpoint based on validation accuracy.
For example the end of the training loop could be re-written to:
```
           # this code block only gets executed when there's no validation dataset
            elif epoch % checkpoint_step == 0 or epoch == num_epochs  # <-- add this bit
                log_or_print(
                    f'this epoch, {epoch}, is a checkpoint epoch.',
                    logger=self.logger, level='info')
                self.save(ckpt_path, epoch)
```

But two checkpoints--and only two--takes up a relatively small amount of space, and makes for easier to read / think about code for the training loop. Also makes it harder to mess up training, testing (code and models) if we make it explicit that there are two types of checkpoints. 
  * E.g., you can be much more confident you are using the `max-val-acc` checkpoint for measuring model accuracy on a test set if you have to literally write out that name when you specify whick checkpoint to use. 
  * And you can clearly test whether each type of checkpoint got saved by looking for the appropriate filename

**Comment by NickleDave** [comment]
@yardencsGitHub I am going ahead and starting on this in a branch but please let me know what you think

**Comment by yardencsGitHub** [comment]
I think this a very good idea.
(also, I was indeed confused by the step-epoch duality and made sure that changing the 'checkpoint_step' and 'vel_step' to 1 indeed saves checkpoint files)

In principle I think one type of users will just want to be able to point to a dataset for training (say, a folder full of .wav files and an annotation file as .csv), instruct 'train with all of this' , and then point to another bunch of .wav files and say 'now predict these'.

Your suggestion perfectly goes in this direction. If it's easy I suggest keeping some potential 'under the hood' capability of tracing the training convergence (decrease of error on training steps and certain validation points)

In terms of using a validation set we should (I think obviously) keep the flexibility of choosing how often to use it. (especially important in large validation sets.. e.g. 1000sec in canaries)

**Comment by NickleDave** [comment]
Addressed in #161
